pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
+                    if valid and mask_longest_chunk < longest_chunk:
+                        cuts = mask_cuts
+                        longest_chunk = mask_longest_chunk
+                if longest_chunk != length:
+                    return [line[a + 1 : b + 1] for a, b in zip([-1, *cuts], cuts)]
+                return [line[i:i + int(length / 3 + 1)] for i in range(0, length, int(length / 3 + 1))]
+
+            if fargs_format == ['*']:
+                return ', '.join(['print *', *fargs]) + '\n'
+            separator = self._print_LiteralString(LiteralString(sep))
+            args_formatting = ''
+            args_code = ''
+            for a_f, a_c in zip(fargs_format, fargs):
+                if a_f == 'A' and len(a_c) > 30 and a_c[0] == '\'' and a_c.count('\'') == 2:
+                    chunks = split_long_argument(a_c)
+                    whole_format = ' '.join(['A'] * len(chunks))
+                    arg_concat = ' , '.join('\'{}\''.format(chunk) for chunk in chunks)
+                    args_code += ('' if args_formatting == ''
+                                  else (' , ' + separator + ' , ' if separator != '' else ' , ')) + arg_concat
+                    args_formatting += ('' if args_formatting == ''
+                                        else (' A ' if separator != '' else ' ')) + whole_format
+                else:
+                    args_code += ('' if args_formatting == ''
+                                  else (' , ' + separator + ' , ' if separator != '' else ' , ')) + (a_c if a_c != '' else '\'\'')
+                    args_formatting += ('' if args_formatting == ''
+                                        else (' A ' if separator != '' else ' ')) + a_f
+            if args_formatting == '':
+                return "write(*, '(A)', advance=\"no\") {}\n".format(self._print_LiteralString(LiteralString(end)))
+            if end != '':
+                return "write(*, '({} A)', advance=\"no\") {} , {}\n".format(args_formatting, args_code, self._print_LiteralString(LiteralString(end)))
+            return "write(*, '({})', advance=\"no\") {}\n".format(args_formatting, args_code)
+
+        if len(orig_args) == 0:
+            return formatted_args_to_print(args_format, args, end)
+
+        for i, f in enumerate(orig_args):
             if f.keyword:
                 continue
             else:
                 f = f.value
             if isinstance(f, str):
-                args.append("'{}'".format(f))
+                arg_format, arg = self.get_print_format_and_arg(f)
+                args_format.append(arg_format)
+                args.append(arg)
             elif isinstance(f, PythonTuple):
-                for i in f:
-                    args.append(self._print(i))
+                arg_format, arg = self.get_print_format_and_arg(f)
+                args_format.append(arg_format)
+                args.append(arg)
```

Comment:
We seem to have lost the loop here, are you sure that this doesn't pose any problems? I don't see any handling of tuples in the function `get_print_format_and_arg`

Reply:
You are right! something is not right here. Could you please suggest some python source code that will enter here?
Didn't your last PR render this part obsolete?
`a = (5, 'one',3)
print(a)
`
this code indicates that f is an instance of `FunctionCallArgument(InhomogeneousTupleVariable('a',dtype=NativeGeneric()))`.

Reply:
Hmm, are you sure? I thought that the line `f = f.value` unpacked the `FunctionCallArgument` so that at this point `f` is an `InhomogeneousTupleVariable`. To print `a` we need to print each element individually, so something like:
```
for i in f:
    arg_format, arg = self.get_print_format_and_arg(i)
    args_format.append(arg_format)
    args.append(arg)
```